tweaks to dartdoc hover rendering

devoncarew opened the pull request:
- remove double blank lines from the the dartdoc tooltip text
- remove the leading `Signature` text
- convert symbol references to markdown code format (`[foo]` ==> ``` `foo` ```)

@alexander-doroshko 

<img width="487" alt="screen shot 2017-03-29 at 1 28 56 pm" src="https://cloud.githubusercontent.com/assets/1269969/24483163/02cd6b4e-14ab-11e7-8c56-9ec85f686cfe.png">

<img width="487" alt="screen shot 2017-03-29 at 6 11 16 pm" src="https://cloud.githubusercontent.com/assets/1269969/24483184/27c22d4a-14ab-11e7-8aa2-b4d06ee9e4f6.png">

devoncarew commented:
It looks like this does break several unit tests - taking a look.

devoncarew commented:
- added fixes for DartServerDocUtilTest and DartDocUtilTest

alexander-doroshko commented:
Brackets are sometimes a part of hyperlink markup. With this change, we loose for example a link in the `DateTime` class, its last line is 
```
the [intl](http://pub.dartlang.org/packages/intl) package.
```

Also smth is wrong with `Object` docs:
![image](https://cloud.githubusercontent.com/assets/4499789/24523246/1645ba46-159b-11e7-803d-ac18c6d68ed5.png)

devoncarew commented:
OK, will update the regex.

devoncarew commented:
Updated. I removed the handling of `[foo]`, but kept some of the whitespace improvements. It looks like the markdown library in use does not correctly handle some `[foo](links)` in any case - the rendering of Object doesn't look correct on master.
